include/trompeloeil.hpp

```diff
@@ -2714,6 +3167,35 @@ namespace trompeloeil
       return {matcher};
     }
 
+    call_modifier&&
+    null_modifier()
+    {
+      return std::move(*this);
+    }
+
+  private:
+    template <
+      typename Sig,
+      typename H,
+      typename R = decltype(::trompeloeil::default_return<return_of_t<Sig>>())>
+    struct throw_handler_t
```

Comment:
Is there a reason for this to be a template? The template types are already known in the surrounding type and should be accessible.

Reply:
It turns out that `Sig` can be dropped as a template parameter, using `signature` instead.
`R` can be made a type alias inside the `struct`.
But `H` remains as a template parameter of `throw_handler_t` as it comes from the
template parameter `H` of `handle_throw`.